urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/filehandler/PHPDirectoryResource.java

```diff
@@ -0,0 +1,227 @@
+     * @param format       The output format; currently "TXT" and "HTML" are supported (default is "TXT", if format is null). 
+     * @param useFairLocks If set to true the created resource will use fair locks.
+     **/
+    public PHPDirectoryResource( HTTPHandler handler,
+				 CustomLogger logger,
+				 HTTPFileFilter fileFilter,
+				 File dir,
+				 URI requestURI,
+				 UUID sid,
+				 String format,
+				 
+				 boolean useFairLocks )
+	throws NullPointerException {
+
+	super( handler, logger, fileFilter, dir, requestURI, sid, useFairLocks );
+
+	this.outputFormat             = format;
+    }
+
+
+    // --- BEGIN --------------------- AbstractDirectoryResource implementation ------------------------
+    /**
+     * This method is designated to build the data for the directory listing.
+     *
+     * Subclasses must implement this method and write the generated data into the given output stream.
+     *
+     *
+     * @param sid The current session's ID.
+     * @param out The output stream to write the list data to.
+     * @throws IOException If any IO errors occur.
+     **/
+    public void generateDirectoryListing( UUID sid,
+					  OutputStream out )
+	throws IOException {
+
+	
+	FileHandler phpHandler = this.getHTTPHandler().getFileHandlerByExtension( ".php" );
+	
+	try {
+	    // Create some dummy values
+
+	    HTTPHeaders dummyHeaders = new HTTPHeaders();
+	    //URI dummyURI             = new URI();
+	    String newPath           = "/system/directory.listing.php";
+	    String newQuery          = "requestURI=" + this.getRequestURI().getPath();
+	    newQuery += "&absoluteDirectoryPath=" + this.getDirectoryFile().getAbsolutePath();
+	    URI dummyURI = new URI( this.getRequestURI().getScheme(), 
+				    this.getRequestURI().getAuthority(), // String authority, 
+				    newPath, // this.getRequestURI().getPath(),      // String path, 
+				    newQuery,   // String query, 
+				    this.getRequestURI().getFragment()  
+				    );
+	    File directory = new File( this.getHTTPHandler().getDocumentRoot(), "system/directory.listing.php" );
+	    System.out.println( dummyURI );
+	    
+	    Resource resource = phpHandler.process( sid,
+						    dummyHeaders, // headers,
+						    null, // postData,
+						    directory, // this.getDirectoryFile(), //file,
+						    dummyURI     //requestURI 
+						    );
+	    resource.open( true ); // Open in read-only mode
+	    InputStream in = resource.getInputStream();
+
+	    //int b;
+	    //while( (b = in.read()) != -1 )
+	    //System.out.println( (char)b );
+
+	    long transferredBytes = CustomUtil.transfer( in, 
+							 out, 
+							 -1, // Long.MAX_VALUE,
+							 256 // buffer size
+							 );
+
+	    resource.close();	    
+	    
+	    //System.out.println( "gak" );
+	    //out.write( new String("This is a test.").getBytes() );
+	    //out.flush();
+
+	} catch( URISyntaxException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 167, 176, 185.